pkg/image/prune/prune.go

```diff
@@ -260,21 +265,6 @@ func addImagesToGraph(g graph.Graph, images *imageapi.ImageList, algorithm prune
 	for i := range images.Items {
 		image := &images.Items[i]
 
-		glog.V(4).Infof("Examining image %q", image.Name)
-
-		if !algorithm.allImages {
-			if image.Annotations[imageapi.ManagedByOpenShiftAnnotation] != "true" {
-				glog.V(4).Infof("Image %q with DockerImageReference %q belongs to an external registry - skipping", image.Name, image.DockerImageReference)
-				continue
-			}
-		}
-
-		age := metav1.Now().Sub(image.CreationTimestamp.Time)
-		if !algorithm.pruneOverSizeLimit && age < algorithm.keepYoungerThan {
-			glog.V(4).Infof("Image %q is younger than minimum pruning age, skipping (age=%v)", image.Name, age)
-			continue
-		}
-
```

Comment:
So if these images were not added to the graph, the algorithm could remove blobs referenced by them, could it not?

Reply:
@miminar yes. The layers will not have links in the graph.

Reply:
That's a serious bug. And we need to back-port this pretty far. The mystery of blobs disappearing from the registry is solved!

Reply:
Could you please capture the case of layers being removed when they should be kept in some unit/extended test? This is pretty important gap we should cover.

Reply:
![](http://e.lvme.me/ejpklmp.jpg)

Reply:
@miminar I'm on it. BZ 1487408